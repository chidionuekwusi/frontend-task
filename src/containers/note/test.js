import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Note from "./index";
import NotesContext from "../../contexts/notes";

describe("<Note />", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <NotesContext.Provider
        value={{
          notes: { "1": { title: "Note", content: "Content" } }
        }}
      >
        <Note match={{ params: { id: "1" } }} />
      </NotesContext.Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("can edit a note", () => {
    let note = { title: "Fake", content: "Content", id: "1" };
    const context = {
      notes: { "1": note },
      edit: (id, _note) => {
        if (id == _note.id) {
          note = _note;
          context.notes[id] = _note;
        }
      },
      get: () => {
        return [note];
      }
    };
    const component = mount(
      <NotesContext.Provider value={context}>
        <Note match={{ params: { id: "1" } }} history={{ replace: () => {} }} />
      </NotesContext.Provider>
    );
    const title = "Something nice";
    const content = "Something great";
    component.find("input").simulate("change", { target: { value: title } });
    component
      .find("textarea")
      .simulate("change", { target: { value: content } });
    component.find("button").simulate("click", {});
    expect(note.title).toBe(title);
    expect(note.content).toBe(content);
  });
});
