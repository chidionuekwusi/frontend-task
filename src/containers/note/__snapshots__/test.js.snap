// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> renders correctly 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      placeholder="Title"
      value="Note"
    />
    <textarea
      onChange={[Function]}
      placeholder="Note"
      value="Content"
    />
  </div>
  <button
    onClick={[Function]}
  >
    <i
      className="fas fa-save"
    />
  </button>
</div>
`;
